x/cronos/keeper/precompiles/relayer.go

```diff
@@ -60,18 +64,31 @@ func (bc *RelayerContract) Address() common.Address {
 }
 
 // RequiredGas calculates the contract gas use
-func (bc *RelayerContract) RequiredGas(input []byte) uint64 {
-	// base cost to prevent large input size
-	baseCost := uint64(len(input)) * bc.kvGasConfig.WriteCostPerByte
+func (bc *RelayerContract) RequiredGas(input []byte) (gas uint64) {
 	if len(input) < prefixSize4Bytes {
 		return 0
 	}
+	intrinsicGas, err := core.IntrinsicGas(input, nil, false, true, true)
+	if err != nil {
+		return 0
+	}
 	prefix := int(binary.LittleEndian.Uint32(input[:prefixSize4Bytes]))
 	requiredGas, ok := relayerGasRequiredByMethod[prefix]
-	if ok {
-		return requiredGas + baseCost
+	if !ok {
+		requiredGas = 0
+	}
+	// base cost to prevent large input size
+	baseCost := uint64(len(input)) * authtypes.DefaultTxSizeCostPerByte
```

Comment:
would a simple `len(input) * a + b` formula has as good approximation?

Reply:
you mean ignore intrinsic like old method right? we'll get `(24196 * 10 + 0) + 405232 - 392132 = 255060` diff



90 validtors | cosmos | evm | adjust | current | input | intrinsic | update client | 2nd method
-- | -- | -- | -- | -- | -- | -- | -- | --
MsgUpdateClient+MsgConnectionOpenTry | 392132 | 405232 | 590472 | 1611456 | 24196 | 405232 | 429428 | 834660

Reply:
pick a better `a` ofc, I assume intrinsic gas is more or less linearly correlated with input length.

Reply:
Actually intrinsic varies with input, let's use TransientGasConfig 3 as before, we'll have negative with smaller nodes `(3684 * 10 + 0) + 77712 - 187154 = -72602`


2 validtors | cosmos | evm | adjust | current | input | intrinsic | update client | 2nd method
-- | -- | -- | -- | -- | -- | -- | -- | --
MsgUpdateClient+MsgConnectionOpenTry | 187154 | 187202 | 232850 | 808302 | 3684 | 77712 | 81396 | 159108

Reply:
Solve this equation:
```
24196 * a + b + 405232 = 392132
3684 * a + b + 77712 = 187154
```

we get `a ~= -6, b ~= 131451`

Reply:
yea, require [0](https://github.com/crypto-org-chain/cronos/blob/7b3c0f4a39bde4db132340e12b300326725b0092/x/cronos/keeper/precompiles/relayer.go#L90) in that case